PhoneBook.Doc415/DataAccess.cs

```diff
@@ -0,0 +1,52 @@
+﻿using PhoneBook.Doc415.Context;
+using PhoneBook.Doc415.Models;
+
+namespace PhoneBook.Doc415;
+
+internal class DataAccess
+{
+    public void AddContact(string _name, string _email, string _phone, string _title)
+    {
+        Contact newContact= new Contact();
+        newContact.Name= _name;
+        newContact.Email= _email;
+        newContact.Title= _title;
+        newContact.PhoneNumber= _phone;
+
+        try
+        {
+            var db = new PhoneBookContext();
+            db.Contacts.Add(newContact);
+            db.SaveChanges();
+        }
+        catch (Exception ex)
+        {
+            Console.WriteLine($"There was an error saving contact: {ex.Message}");
+        }
+    }
+
+    public List<Contact> GetContacts()
+    {
+        using var db = new PhoneBookContext();
+        return db.Contacts.OrderBy(x=>x.Name).ToList();
+    }
+
+    public void DeleteContact(Contact todelete)
+    {
+        using var db = new PhoneBookContext();
+        db.Contacts.Remove(todelete);
+        db.SaveChanges();
+    }
+
+    public void UpdateContact(string _name, string _title, string _phone,string _email,Contact toUpdate)
```

Comment:
why passing `Contact toUpdate`? I think what Update interface should look like it's to have `ContactId`. Then you retrieve Contactid, update and save it. It feels strange that you need to retrieve yourself (as a caller) first object and then pass it for updating. 

Then why not accept only `Contact toUpdate` because caller might update it as well

Reply:
İ have already get that obj to memory in ui class to see its properties. I am just passing it . No need to query db to get it again. It is already too busy.

Reply:
If you show in UI your DB models, then you don't have enough separation of concerns. 
You will have problems when you want to show something more then model (an union of two db models for example) and then change them accordingly. 

> No need to query db to get it again. It is already too busy.

My comment it's not about performance. If you worry so much about performance, then even more you should disable `Tracking` of models from EF for all objects that won't be updated. 

For trackable object create sort of cache mechanism and then update it like that if you don't want to query it

Reply:
> If you show in UI your DB models, then you don't have enough separation of concerns.

user need to select a contact to update or delete??
how to present the list to user??

UI asks to dataaccess to get contact list, show it to user
user selects one of them , ui returns it to dataaccess to modify

What kind of structure do you suggest?

Reply:
There are multiple ways to approach it. The cleanest solution would be to go with "ports and adapters" architecture pattern, but I think for this it's overkill.
what you can do it's to have 2 different sets of classes
One models for DB. One models for ui Then you create Data Access Layer and some "Service Layer" that does validation and mapping from DB models into UI models